package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -60,10 +84,112 @@ def to_json(self) -> str:
         database.commit()
+        self.sync()
+
+    def _upload(self):
+        """Uploads the session store database file to the specified remote path on the cloud storage."""
+        # Connect to the remote file system
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        db_name = _get_dbname()
+        fs = fsspec.filesystem(protocol)
+        # Open the local database file and upload it to the remote path
+        with open(self.location, "rb") as file:
+            with fs.open(f"{self._remote_path}/{db_name}", "wb") as s3f:
+                s3f.write(file.read())
+
+    def _download(self) -> List[str]:
+        """Downloads all the session store database files from the specified remote path on the cloud storage
+        to your local project.
+        Note: All the database files are deleted after they are merged to the main session_store.db.
+
+        Returns:
+        A list of local filepath in string format for all the databases
+
+        """
+
+        # Connect to the remote file system
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        fs = fsspec.filesystem(protocol)
+        # Find all the databases at the remote path
+        databases = fs.glob(f"{self._remote_path}/*.db")
+        databases_location = []
+        # Download each database to a local filepath
+        for database in databases:
+            database_name = Path(database).name
+            with fs.open(database, "rb") as file:
+                db_data = file.read()
+                db_loc = f"{self._path}/{database_name}"
+                with open(db_loc, "wb") as temp_db:
+                    temp_db.write(db_data)
+            databases_location.append(db_loc)
+        # Return the list of local filepaths
+        return databases_location
+
+    def _merge(self, databases_location: List[str]):
+        """Merges all the session store databases stored at the specified locations into the user's local session_store.db
+
+        Notes:
+        - This method uses multiple SQLAlchemy engines to connect to the user's session_store.db and to all the other downloaded dbs.
+        - It is assumed that all the databases share the same schema.
+        - In the version 1.0 - we only merge the runs table which contains all the experiments.
+        - The downloaded database files are deleted after it's runs are merged with the user's local session_store.db
+
+        Args:
+            database_location:  A list of local filepath in string format for all the databases
+
+        """
+
+        # Connect to the user's local session_store.db
+        engine, session_class = create_db_engine(self.location)
+        Base.metadata.create_all(bind=engine)
+        database = next(get_db(session_class))
+
+        temp_engine = None
+        # Iterate through each downloaded database
+        for db_loc in databases_location:
+            if temp_engine:
+                temp_engine.dispose()
```

Comment:
What is this for?

Reply:
If there's a connection already open, it closes it. Just to make sure we don't have too many open connections